src/openapi/streaming/connection/transport/websocket-transport.js

```diff
@@ -25,8 +25,50 @@ const DEFAULT_RECONNECT_LIMIT = 10;
 
 const NOOP = () => {};
 
+let maxCallstackSize;
+
 // -- Local methods section --
 
+function getMaxCallstackSize() {
+    if (maxCallstackSize) {
+        return maxCallstackSize;
+    }
+
+    let index = 0;
+    function recurse() {
+        index++;
+        recurse();
+    }
+
+    try {
+        recurse();
+    } catch (e) {
+        maxCallstackSize = index;
+    }
+
+    return maxCallstackSize;
+}
+
+function getJSONPayloadString(payloadBuffer) {
+    const chunkSize = getMaxCallstackSize();

```

Comment:
I don't think we should attempt to get a maximum call stack - its an expensive operation and dangerous - I've seen older browsers crash and sometimes be unable to catch the exception.

I compared some techniques and even setting the chunk size to 150 its not that much difference from whatever max call size is. at 350 chunk size its 20% slower.. so a pick a number thats definitely very safe - for instance I'd be happy with 1000 ?

![image](https://user-images.githubusercontent.com/309321/84470321-64cbea00-ac83-11ea-86ee-c66076fa9a04.png)

Reply:
https://jsperf.com/yugk-fromcharcode
1000 seems good enough